src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -840,7 +840,26 @@ public void execute() throws GitException, InterruptedException {
 
                 args.add(fastForwardMode);
                 args.add(rev.name());
-                launchCommand(args);
+
+                /* See JENKINS-45228 */
```

Comment:
Right after merging this pull request these 2 comment lines get outdated.
Better put a comment like "If there is a remote URL, the merge might require a credential because LFS might be used which connects to the remote." at the top of the first if block to describe shortly why this is done.